src/movies.js

```diff
@@ -1,16 +1,89 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(array){
+    return array.map((movie) => movie.director);
+/*const allDirectors = array.map(movie => { 
+    const container = movie.director
+    return container})
+    return allDirectors*/
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+Array.from(new Set(getAllDirectors(movies)))
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(array){
+   
+return (array.filter( item =>
+    item.director === "Steven Spielberg" &&
+    item.genre.includes('Drama')
+    )).length
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(array){
+    if(!array.length){
+        return 0
+    }
+    const allRatings = array.map((movie) => movie.rate);
+
+    let total = allRatings.reduce(function(sum, item){
+        return sum + item
+    })
+
+    return Math.round((total/array.length)*100)/100
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(array){
+    const drama = array.filter( item =>
+    item.genre.includes('Drama')
+    )
+    if (!drama.length){
+        return 0
+    }
+
+    const allRatings = drama.map(movie => { 
+    const container = movie.rate
+    
+    return container})
+
+    let total = allRatings.reduce(function(sum, item){
+        return sum + item
+    })
+    return Math.round((total/drama.length)*100)/100
+}
 
```

Comment:
It would have been more efficient to make use of your `ratesAverage` function

`function dramaMoviesRate(array){
    const drama = array.filter( item =>
    item.genre.includes('Drama')
    )
    return ratesAverage(drama);
}`